modules/wss4j/src/org/apache/ws/security/processor/SAML2TokenProcessor.java

```diff
@@ -97,22 +98,22 @@ public void handleToken(Element elem, Crypto crypto, Crypto decCrypto, CallbackH
     public Assertion buildAssertion(Element elem) throws WSSecurityException {
         Assertion samlAssertion;
         try {
-            DefaultBootstrap.bootstrap();
+            SAMLInitializer.doBootstrap();
```

Comment:
Is the bootstrap being done per request? IIRC this should be done only once. So could have done at the constructor or somewhere which handle the initialization logic

Reply:
initialization is done as done in the previous implementation. We can improve the code in SAMLInitializer to execute if is has not been done before.

Reply:
Is it already implemented in 'SAMLInitializer'? If not please do that as well.

Reply:
+1. Will update the SAMLInitializer and upgrade the version